src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -123,7 +123,6 @@ class ProtobufPlugin implements Plugin<Project> {
     }
 
     private static void linkGenerateProtoTasksToTask(Task task, GenerateProtoTask genProtoTask) {
-      task.dependsOn(genProtoTask)
       task.source genProtoTask.getOutputSourceDirectorySet().include("**/*.java", "**/*.kt")
```

Comment:
We can pass the task as an argument to the `source` function. dependOn will be added implicitly.

From docs:
[source](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceTask.html#org.gradle.api.tasks.SourceTask:source(java.lang.Object[]))(sources)	
Adds some source to this task. The given source objects will be evaluated as per [Project.files(java.lang.Object[])](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:files(java.lang.Object[])).

[ConfigurableFileCollection](https://docs.gradle.org/current/javadoc/org/gradle/api/file/ConfigurableFileCollection.html) Project.files([Object](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html)... paths)
You can pass any of the following types to this method:
<...>
A [Task](https://docs.gradle.org/current/dsl/org.gradle.api.Task.html). Converted to the task's output files. The task is executed if the file collection is used as an input to another task.